pkg/server/sandbox_list_test.go

```diff
@@ -0,0 +1,210 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/net/context"
+
+	"github.com/containerd/containerd/api/types/container"
+
+	"k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
+
+	"github.com/kubernetes-incubator/cri-containerd/pkg/metadata"
+	servertesting "github.com/kubernetes-incubator/cri-containerd/pkg/server/testing"
+)
+
+func TestToCRISandbox(t *testing.T) {
+	config := &runtime.PodSandboxConfig{
+		Metadata: &runtime.PodSandboxMetadata{
+			Name:      "test-name",
+			Uid:       "test-uid",
+			Namespace: "test-ns",
+			Attempt:   1,
+		},
+		Labels:      map[string]string{"a": "b"},
+		Annotations: map[string]string{"c": "d"},
+	}
+	createdAt := time.Now().UnixNano()
+	meta := &metadata.SandboxMetadata{
+		ID:        "test-id",
+		Name:      "test-name",
+		Config:    config,
+		CreatedAt: createdAt,
+		NetNS:     "test-netns",
+	}
+	state := runtime.PodSandboxState_SANDBOX_READY
+	expect := &runtime.PodSandbox{
+		Id:          "test-id",
+		Metadata:    config.GetMetadata(),
+		State:       state,
+		CreatedAt:   createdAt,
+		Labels:      config.GetLabels(),
+		Annotations: config.GetAnnotations(),
+	}
+	s := toCRISandbox(meta, state)
+	assert.Equal(t, expect, s)
+}
+
+func TestFilterSandboxes(t *testing.T) {
+	c := newTestCRIContainerdService()
+
+	testSandboxes := []*runtime.PodSandbox{
+		{
+			Id:       "1",
+			Metadata: &runtime.PodSandboxMetadata{Name: "name-1", Uid: "uid-1", Namespace: "ns-1", Attempt: 1},
+			State:    runtime.PodSandboxState_SANDBOX_READY,
+		},
+		{
+			Id:       "2",
+			Metadata: &runtime.PodSandboxMetadata{Name: "name-2", Uid: "uid-2", Namespace: "ns-2", Attempt: 2},
+			State:    runtime.PodSandboxState_SANDBOX_NOTREADY,
+			Labels:   map[string]string{"a": "b"},
+		},
+		{
+			Id:       "3",
+			Metadata: &runtime.PodSandboxMetadata{Name: "name-2", Uid: "uid-2", Namespace: "ns-2", Attempt: 2},
+			State:    runtime.PodSandboxState_SANDBOX_READY,
+			Labels:   map[string]string{"c": "d"},
```

Comment:
What do you mean? :)
TODO for filtering? Or fill the fields?

`TestToCRISandbox` is the test to test all fiedls.